sematic/resolvers/silent_resolver.py

```diff
@@ -48,5 +67,30 @@ def _end_inline_execution(self, future_id) -> None:
         """Callback called at the end of an inline execution."""
         pass
 
+    def _future_did_terminate(self, future):
+        self._deactivate_resources(future)
+
     def _wait_for_scheduled_runs(self) -> None:
         pass
+
+    def _register_external_resources(self, future) -> None:
+        for external_resource in future.props.external_resources:
+            used_resource_id = external_resource.id
+            for other_future in self._futures:
+                if other_future.id == future.id:
+                    continue
+                if any(
+                    r.id == used_resource_id
+                    for r in other_future.props.external_resources
+                ):
+                    # See https://github.com/sematic-ai/sematic/issues/382
+                    use_1 = other_future.calculator.__name__
+                    use_2 = future.calculator.__name__
+                    raise RuntimeError(
```

Comment:
Same TODO comment.

Reply:
a couple lines above there's a link to a ticket